components/locale/src/fallback/algorithms.rs

```diff
@@ -56,36 +51,23 @@ impl<'a> LocaleFallbackerWithConfig<'a> {
                         )
                         .unwrap_or(default_script)
                 {
-                    locale.set_script(None);
+                    langid.script = None;
                 }
             } else if script == default_script {
-                locale.set_script(None);
+                langid.script = None;
             }
         }
-        // 3. Remove irrelevant extension subtags
-        locale.retain_unicode_ext(|key| {
-            match *key {
-                // Always retain -u-sd
-                SUBDIVISION_KEY => true,
```

Comment:
Thought/Issue: We don't currently use `-u-sd` but it would be nice to support it in LocaleFallbacker for future compatibility

Reply:
This is a new requirement, please don't block the PR on this.

Reply:
I won't block the PR on it but it's not a new requirement; the old fallbacker supported -u-sd so it's a regression in behavior that it doesn't support it now and doesn't have a clear path to supporting it...

Reply:
We had a design discussion in March to drop unicode keywords from the fallbacker and this was not brought up since then. This will require a lot of rethinking.